metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_queries.py

```diff
@@ -696,6 +696,9 @@ def _build_enriched_query_log_query(
     JOIN filtered_access_history a USING (query_id)
 )
 SELECT * FROM query_access_history
+-- Our query aggregator expects the queries to be added in chronological order.
+-- It's easier for us to push down the sorting to Snowflake/SQL instead of doing it in Python.
+ORDER BY start_time ASC
```

Comment:
QUERY_START_TIME ?

Reply:
I checked if we are doing the sorting in _eg_ bigquery and yes, we do
https://github.com/datahub-project/datahub/blob/ac13f255e6f2dd6fc95e2d25ff610b842a2d0aa9/metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py#L503

Are there other sources to validate?

Reply:
Snowflake is getting fixed here, bq is all good. For redshift, we actually register temp tables as a preprocessing step, so it's fine that the other queries don't have an `order by`

Reply:
What if we add a validation in sql parsing aggregator? we may track start query time as queries are added to the aggregator and _eg_ we may raise a warning if the start query time is not increasing for every addition